### PR TITLE
Write transactions on commit

### LINKED ISSUE #2009: Write transactions on block commit
### Description
The transactions in a block are delivered by Tendermint using `deliver_tx` and in the current implementation they are [stored when they are received](https://github.com/bigchaindb/bigchaindb/blob/tendermint/bigchaindb/tendermint/core.py#L88). In case there is any crash during `deliver_tx`, [`end_block`](https://github.com/bigchaindb/bigchaindb/blob/tendermint/bigchaindb/tendermint/core.py#L92) or [`commit`](https://github.com/bigchaindb/bigchaindb/blob/tendermint/bigchaindb/tendermint/core.py#L110) the database will be left in an inconsistent state i.e. Tendermint will try to replay the transaction which will only result in failure as some transactions may already be present in the database.

### Approach
A possible approach to resolve this would be to write transactions during [`commit`](https://github.com/bigchaindb/bigchaindb/blob/tendermint/bigchaindb/tendermint/core.py#L110).

/cc @vrde @sbellem

### PR DESCRIPTION
**Pull request by kansi**
Fixes #2009

### CODE REVIEW

File: bigchaindb/backend/query.py [hunk 1]
```diff
@@ -33,6 +33,20 @@ def store_asset(connection, asset):
     raise NotImplementedError
 
 
+@singledispatch
+def store_assets(connection, assets):
+    """Write an asset to the asset table.
```
Comment (ttmc) [inline bigchaindb/backend/query.py]: Write a list of assets to the asset table.

File: bigchaindb/backend/query.py [hunk 2]
```diff
@@ -33,6 +33,20 @@ def store_asset(connection, asset):
...
+
+    Args:
+        assets ([dict]): list of assets.
```
Comment (ttmc) [inline bigchaindb/backend/query.py]: Is the type a dict or a list of dicts?

The docstring for `store_metadata` (or whatever it gets renamed as) should also say that it stores a list.

File: bigchaindb/backend/localmongodb/query.py [hunk 3]
```diff
@@ -71,6 +80,16 @@ def store_asset(conn, asset):
         pass
 
 
+@register_query(LocalMongoDBConnection)
+def store_assets(conn, assets):
```
[the remark this answers is not in the document]
Comment (ttmc) [reply bigchaindb/backend/localmongodb/query.py]: I wonder why there is no analogue method for storing a single metadata.

[the remark this answers is not in the document]
Comment (kansi) [reply bigchaindb/backend/localmongodb/query.py]: The analogue storing single metadata is achieved by sending a list containing one metadata.

[the remark this answers is not in the document]
Comment (ttmc) [reply bigchaindb/backend/localmongodb/query.py]: Haha, yes, but you could also say the same about `store_transactions` and `store_assets`.

[the remark this answers is not in the document]
Comment (kansi) [reply bigchaindb/backend/localmongodb/query.py]: Yeah. actually I wrote extraction and storage code for metadata and I chose to have one function. The transaction and metadata already had single and bulk storage routines.

[the remark this answers is not in the document]
Comment (ttmc) [reply bigchaindb/backend/localmongodb/query.py]: :+1:

**Pull request merged**